packages/style-engine/class-wp-style-engine.php

```diff
@@ -162,6 +162,14 @@ class WP_Style_Engine {
 					'spacing' => '--wp--preset--spacing--$slug',
 				),
 			),
+			'blockGap' => array(
```

Comment:
Thanks for looking at this one! 

I don't know if it'll help, but I was tried something similar over at https://github.com/WordPress/gutenberg/pull/40955/files#diff-69616e5e4ff1a06362db0b7165c330b00614d4ea0d3a28c80003c0860ab8b93fR176

Reply:
Oh, thank you! I'm sure it will — I was mostly hacking around pretty quickly on this yesterday to see how we may or may not be able to slot things in, and put up the PR to save my work. I have no idea if the approach will work / be viable, but was curious to have a poke around 😄

With this PR I'm hoping to see whether on balance, it's worth attempting to move some of the Layout logic (specifically `blockGap`) over to the style engine, or if it's just easier to keep it as part of the Layout logic, and then duplicate the parsing of CSS variables.

The end goal is so that once the presets-based UI for spacing is ready, we are then able to roll it out to block gap/spacing in addition to padding and margins. 🤞